lib/test_track_rails_client/assignment_helper.rb

```diff
@@ -1,19 +1,39 @@
 module TestTrackRailsClient::AssignmentHelper
-  def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+  def stub_test_track_assignments(assignment_registry) # rubocop:disable Metrics/AbcSize
     raise "Cannot stub test track assignments when TestTrack is enabled" if TestTrack.enabled?
 
-    split_registry = {}
-    assignments = []
+    schema = TestTrackRailsClient::Schema.new
 
+    assignments = []
     assignment_registry.each do |split_name, variant|
-      assignment_registry[split_name] = variant.to_s
-      split_registry[split_name] = { variant => 100 }
+      raise "Variant #{variant} for #{split_name} must be a string or symbol" unless [String, Symbol].include?(variant.class)
+      raise "Split #{split_name} does not exist in #{schema.file_path}" unless schema.splits.include?(split_name)
```

Comment:
Now that we have access to the schema, we can validate that the splits we're stubbing actually exists.